### PR TITLE
WT-9489 Update search near to work with bounds for column store

### PR DESCRIPTION
**Pull request by clarissecheah**

### CODE REVIEW

File: test/suite/test_cursor_bound08.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ class test_cursor_bound08(bound_base):
     key_format_values = [
         ('string', dict(key_format='S')),
         # FIXME-WT-9489: Uncomment once column store search near is implemented.
```
Comment (jiechenbo) [inline test/suite/test_cursor_bound08.py:51]: Remove fix me comment

File: test/suite/test_cursor_bound09.py [hunk 2]
```diff
@@ -46,7 +46,7 @@ class test_cursor_bound09(bound_base):
     key_format_values = [
         ('string', dict(key_format='S')),
         # FIXME-WT-9489: Uncomment once column store search near is implemented.
```
Comment (jiechenbo) [inline test/suite/test_cursor_bound09.py:48]: ditto

File: test/suite/test_cursor_bound08.py [hunk 3]
```diff
@@ -212,7 +212,7 @@ def test_bound_perf_stat_scenario(self):
         self.assertGreater(skip_count - prev_skip_count, 50 * 2)
 
         # Test bound api: Test that cursor bound search near traverses less entries perf on lower bounds.
-        prev_skip_count = skip_count
+        prev_skip_count = skip_count
```
Comment (jiechenbo) [inline test/suite/test_cursor_bound08.py:215]: np: remove extra whitespace

File: test/suite/test_cursor_bound08.py [hunk 4]
```diff
@@ -203,7 +203,7 @@ def test_bound_perf_stat_scenario(self):
         # range forward, and then the whole range backwards.
         self.assertGreater(skip_count, 2 * key_count - 200)
 
-        prev_skip_count = skip_count
+        prev_skip_count = skip_count
```
Comment (jiechenbo) [inline test/suite/test_cursor_bound08.py:206]: ditto.

File: src/btree/bt_cursor.c [hunk 5]
```diff
@@ -77,8 +74,13 @@ __btcur_bounds_search_near_reposition(WT_SESSION_IMPL *session, WT_CURSOR_BTREE
...
-      session, cursor, &cursor->key, WT_RECNO_OOB, &key_out_of_bounds, &upper));
+    if (CUR2BT(cursor)->type == BTREE_ROW)
+        WT_RET(__btcur_bounds_contains_key(
+          session, cursor, &cursor->key, WT_RECNO_OOB, &key_out_of_bounds, &upper));
+    else
+        WT_RET(__btcur_bounds_contains_key(
+          session, cursor, NULL, cursor->recno, &key_out_of_bounds, &upper));
+
     if (key_out_of_bounds) {
         __wt_cursor_set_raw_key(cursor, upper ? &cursor->upper_bound : &cursor->lower_bound);
```
Comment (jiechenbo) [inline src/btree/bt_cursor.c:85]: Just confirming, does `__wt_cursor_set_raw_key` set the cursor->recno?

**Review by jiechenbo — COMMENTED** [review]
Mostly looks good, just have a few questions and fixups

File: test/suite/test_cursor_bound08.py [hunk 6]
```diff
@@ -87,7 +87,7 @@ def create_session_and_cursor_timestamp(self):
 
         if (self.evict):
             evict_cursor = self.session.open_cursor(uri, None, "debug=(release_evict)")
-            for i in range(0, 1000):
+            for i in range(100, 1000):
```
Comment (sauclovian-wt) [inline test/suite/test_cursor_bound08.py:90]: Doesn't really matter, but you can use 1 here (it's 0 specifically that's reserved)... most tests now do `for i in range (1, nrows + 1)`.

File: src/btree/bt_cursor.c (hunk 5 above)
Comment (clarissecheah) [reply src/btree/bt_cursor.c:85]: Good question!  `__wt_cursor_set_raw_key` calls `__wt_cursor_set_keyv` internally which sets the cursor->recno if it's column store.

File: test/suite/test_cursor_bound08.py (hunk 6 above)
Comment (clarissecheah) [reply test/suite/test_cursor_bound08.py:90]: That's a good point, thanks for bringing that up. Have changed it as suggested!

**Review by jiechenbo — APPROVED** [review]
LGTM! Nice work 👍

**Review by luke-pearson — APPROVED** [review]
LGTM, nice work!

**Pull request merged**